euphonic/design/05_powder_averaging_discussion.md

```diff
@@ -0,0 +1,318 @@
+-   When sampling a regular sequence of spherical shells, it seems sufficient to average over the points taken at each *q*.
+-   If our samples are placed arbitrarily in space, it is less obvious how to determine these weights
+
+### How do we sample the space to ensure that all of 4-D space is included in the results (avoiding "blind spots")?
+-   This is a known problem and is solved by sampling a lot of points
+-   Consider the topology/features of the space; there are more
+    features intersecting a spherical shell at large *q* than a
+    smaller shell at low *q*. This implies that a greater number of
+    samples is needed as we go further out in *q*.
+
+### How can we do this efficiently?
+
+-   Use symmetry!
+    - The irreducible wedge of the Brillouin zone can be
+      drastically smaller than the full space while containing all
+      the unique information.
+    - Don't try too hard to make origin shifts that squeeze a bit
+      more efficiency from symmetry operations; these benefits
+      diminish as the number of samples increases.
+-   Use caching
+    - The *q*-range in INS measurements passes through many cells
+      of the reciprocal lattice. If possible, re-use computed
+      frequency/qpts in extended zone scheme.
+    - Brille solves some of these problems by transparently
+      re-using appropriate data.
+    - Would be nice to intentionally select long-range *q-points*
+      that already have calculated frequencies so that only S
+      needs recalculating. This might not be commensurate with the
+      desired *q* bins, however.
+
+### How can we check the convergence? Is a "fire-and-forget" approach feasible?
+
+- This could be a substantial project in its own right. To progress
+  the project we should get a baseline method in place and then
+  experiment from there, rather than designing everything in theory
+  first.
+- Fundamentally the goal of the sampling exercise is to perform a
+  numerical integration. In the numerical methods field there has been
+  a lot of work on integration problems, including methods which use
+  adaptive step sizes to obtain a consistent uncertainty when
+  integrating arbitrarily complex functions.
+- This function is not arbitrary and there should be some useful
+  information lurking in the force constants data.
+
+
+### Is the same sampling mesh suitable for the **A** matrix used in Debye-Waller calculation, or should this be a separate q-mesh?
+
+- This point was not discussed in the meeting. It's worth observing
+  that DW should only need sampling within a single Brillouin zone,
+  while S reaches out further in reciprocal space. We'll probably stick with
+  Monkhorst-Pack grids initially.
+
+
+### Can we use statistical methods to quantify uncertainty in the convergence?
+
+- Statistical methods, propogation of error approaches could provide
+  some useful tools
+- Experimentally we can examine uncertainty by comparing different
+  detector contributions to the same bin
```

Comment:
I guess you mean here that we would have multiple detectors at the same 2theta but different phi angles so they would all have the same abs(q) so contribute the same abs(q) bin?

If so, maybe you can change "same bin" to "same abs(q) bin" to make it clearer? (sorry it took me a few seconds to understand).

Reply:
I think that's probably what was meant.

But out of interest, would it also be possible to compare experimental detectors contributing to the same **q** on some instruments? Is there some overlap between detectors with a different 2(theta) at a different time-of-flight?

Reply:
Yes (I think).

So, actually the ToF dependence of each detector represents a 1D path in (**q**, ω), so different detectors *do* contribute to the same **q**, but at different ω. 

In practice what we do with the data is to first convert from ToF to energy transfer ω, and then rebin it in a regular ω grid (the ToF data would probably have been measured in a regular ToF grid which would yield a ragged ω grid).

This then gives data where the **q** bins of each ω-binned strip would differ from ω-bin to ω-bin (because they intersect the 1D (**q**, ω) trajectory of the detectors at different points).

For single-crystal data, the data is stored on disk in this ragged-**q** bin format and rebinned into a regular **q** grid on the fly by the data analysis software (either MSlice or Horace). 

For polycrystalline data, the data is actually rebinned in regular *2-theta* bins (so ragged in abs(q)) and stored on disk in this grid. It is then rebinned into regular abs(q) bins by the data analysis software (MSlice).

To have different detectors with the same **q** *and* ω would be possible on instruments like MARI/MAPS/MERLIN but would need multiple measurements with different incident neutron energy/wavelength. It would not be possible on a TOSCA-like instrument because the scattered neutron energy is fixed by the analyser crystals and cannot be varied from measurement to measurement.